control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -603,6 +622,27 @@ impl ResourceSpecsLocked {
         }
         pending_ops
     }
+
+    /// Get the list of nodes where the replicas of the volume are currently placed.
+    pub(crate) fn get_volume_replica_nodes(
+        &self,
+        registry: &Registry,
+        volume_id: &VolumeId,
+    ) -> Vec<NodeId> {
+        // Map of pool id to the node id.
+        let mut pool_node_map: HashMap<PoolId, NodeId> = HashMap::new();
```

Comment:
don't think we need this temporary? we can just compare using `registry.specs().pools()` ?

Reply:
We can but that would cost some searching iterations in the list. Using a hashmap reduces the search time?